frameworks/zpm.zsh

```diff
@@ -0,0 +1,52 @@
+() {
+
+local -r home_dir=${1}
+
+git clone --recursive https://github.com/zpm-zsh/zpm $home_dir/.zpm
+
+>| ${home_dir}/.zshrc <<\END
+autoload -Uz --                 \
+  git-alias-lookup              \
+  git-branch-current            \
+  git-branch-delete-interactive \
+  git-dir                       \
+  git-ignore-add                \
+  git-root                      \
+  git-stash-clear-interactive   \
+  git-stash-recover             \
+  git-submodule-move            \
+  git-submodule-remove          \
+  mkcd                          \
+  mkpw                          \
+  duration-info-precmd          \
+  duration-info-prexec          \
+  coalesce                      \
+  git-action                    \
+  git-info
```

Comment:
```suggestion
# zpm adds functions to fpath but does not autoload them!
() {
  setopt LOCAL_OPTIONS EXTENDED_GLOB
  local zfunction
  for zfunction in ${HOME}/.zpm/plugins/zimfw--*/functions/^(*~|*.zwc(|.old)|_*|prompt_*_setup)(N-.:t); do
    autoload -Uz -- ${zfunction}
  done
}
```
Please let's make this dynamic. The list can change over time, and there's a typo in prexec which should be preexec.

Reply:
Autoloading everything at the end won't work because it's too late. E.g., `zimfw/asciiship` checks for `git-info` during sourcing.

I've changed the code so that it errors out if the list of functions changes. What do you think?

Alternatively, I can define a function within zshrc that invokes `zpm load` followed by `autoload`. None of the other files under `frameworks/` do that though.

Reply:
We should definitely autoload at the beginning, I was not proposing to change that.

If I were a user of such framework, and had to autoload all the functions, I would do it in a dynamic way. E.g. we recently added git-branch-remote-tracking to the git module, and we don't expect users to care about this now needing to be autoloaded manually somewhere. The purpose of prezto modules is that autoloading should be automatic for users.

Maybe we can simply the code in the beginning of .zshrc to a minimum, like:
```
setopt EXTENDED_GLOB
autoload -Uz -- ${home_dir}/.zpm/plugins/zimfw--*/functions/^(*~|*.zwc(|.old)|_*|prompt_*_setup)(N-.:t)
```
The penalty in time from the static to the dynamic solution in my machine was 0.2ms. Considering how slow my machine is compared to the GitHub Actions machine, this would translate to around 0.165ms in the benchmark generated. I consider this negligible and, even when considering it at the amount it has, a fair penalty given the framework lack of the feature.